### PR TITLE
[Pipeline Viz] Add arrows between stages

### PR DESCRIPTION
**Pull request by emilyzhong**
# Add arrows between stages
Adds arrows between stages, + other misc. UI updates

image

### CODE REVIEW

File: app/assets/src/components/views/PipelineViz.jsx [hunk 1]
```diff
@@ -326,10 +328,7 @@ class PipelineViz extends React.Component {
           <div className={isOpened ? cs.openedStage : cs.hidden}>
             <div className={cs.graphLabel}>
               {stageName}
-              <RemoveIcon
-                className={cs.closeButton}
-                onClick={() => this.toggleStage(i)}
-              />
+              <RemoveIcon onClick={() => this.toggleStage(i)} />
```
Comment (lvreynoso) [inline app/assets/src/components/views/PipelineViz.jsx]: I believe you can just write `onClick={this.toggleStage(i)}`

File: app/assets/src/components/views/pipeline_viz.scss [hunk 2]
```diff
@@ -4,38 +4,62 @@
...
   border-radius: 8px;
   padding: 12px;
-  width: 120px;
+  width: 250px;
   text-align: center;
   font-weight: $font-weight-bold;
   cursor: pointer;
 }
 
+.stageArrow {
```
Comment (lvreynoso) [inline app/assets/src/components/views/pipeline_viz.scss]: Would this not be better as a React component that lets you pass in the width of the arrow as a prop? Or will stage arrows always be static?

Comment (emilyzhong) [reply app/assets/src/components/views/pipeline_viz.scss]: For now stage arrows are static, and are only used in the context of this specific part of the graph so not sure if it's worth creating a whole separate component

File: app/assets/src/components/views/PipelineViz.jsx [hunk 3]
```diff
@@ -342,6 +341,19 @@ class PipelineViz extends React.Component {
       );
     });
 
+    stageContainers = stageContainers.reduce((containers, stage, i) => {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/PipelineViz.jsx]: nit: you can just call reduce on the output of the map and make `stageContainers` const again.

File: app/assets/src/components/views/PipelineViz.jsx [hunk 4]
```diff
@@ -311,7 +313,7 @@ class PipelineViz extends React.Component {
   }
 
   render() {
-    const stageContainers = this.stageNames.map((stageName, i) => {
+    let stageContainers = this.stageNames.map((stageName, i) => {
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/PipelineViz.jsx]: This is getting harder read. Can you split into
* a function that render container
* a function that render stage arrow
* render that then loops through the stages and calls `renderStageArrow` for the first stage and `renderContainer` for all the others
?

File: app/assets/src/components/views/pipeline_viz.scss [hunk 5]
```diff
@@ -4,38 +4,62 @@
...
-  margin: 24px;
+  margin: 24px 0;
 }
 
 .stageButton {
   background-color: $light-grey;
   border-radius: 8px;
   padding: 12px;
-  width: 120px;
+  width: 250px;
```
Comment (tfrcarvalho) [inline app/assets/src/components/views/pipeline_viz.scss]: Can this be automatic width base on nowrap text?
Or is it supposed to have the same width for all the stages?
If the latter, can you make sure the text wraps if ever necessary `word-break: break-word`.

Comment (emilyzhong) [reply app/assets/src/components/views/pipeline_viz.scss]: The mocks have them all at the same width but I can ask jenn about it on Monday to confirm!

**Pull request merged**